main.js

```diff
@@ -160,4 +160,151 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+// OPDRACHT 1
+//opdracht 1a
+const televisionTypes = inventory.map((television) => {
+  return television.type;
+})
+console.log(televisionTypes);
+
+//opdracht 1b
+const soldOut = inventory.filter((television) =>{
+  if (television.originalStock - television.sold === 0){
+    return true;
+  }
+})
+console.log(soldOut);
+
+//opdracht 1c
+const containsAmbi = inventory.filter((television) =>{
+  if (television.options.ambiLight === true ){
+    return true;
+  }
+})
+console.log(containsAmbi);
+
+//opdracht 1d
+
+function sortingTv (){
+inventory.sort((a,b) => {
+  return a.price-b.price;
+})
+console.log(inventory);
+}
+// sortingTv(inventory); -->  Even uitgezet, want anders wordt Array gesorteerd
+
+
+//OPDRACHT 2
+//opdracht 2a
+let soldTelevisions=0;
+for (let i=0; i<inventory.length; i++){
+ soldTelevisions = soldTelevisions + inventory[i].sold;
+}
+console.log("Er zijn " + soldTelevisions + " verkocht");
+
+//opdracht 2b
+const element = document.getElementById("soldTVs");
+element.textContent = `Er zijn ${soldTelevisions} televisies verkocht`;
+
+//opdracht 2c
+let boughtTelevisions=0;
+for (let i=0; i<inventory.length; i++){
+  boughtTelevisions = boughtTelevisions + inventory[i].originalStock;
+}
+console.log("Er zijn " + boughtTelevisions + " ingekocht");
+
+// //opdracht 2d
+const element2 = document.getElementById("boughtTVs");
+element2.textContent = `Er zijn ${boughtTelevisions} televisies oorspronkelijk ingekocht`;
+
+//opdracht 2e
+let televisionsNeedToBeSold=0;
+for (let i=0; i<inventory.length; i++){
+  televisionsNeedToBeSold = televisionsNeedToBeSold + (inventory[i].originalStock- inventory[i].sold);
+}
+console.log("Er zijn " + televisionsNeedToBeSold + " nog over");
+
+const element3 = document.getElementById("leftOver");
+element3.textContent = `Er zijn ${televisionsNeedToBeSold} televisies op voorraad`;
+
+//OPDRACHT3
+//opdracht3a
+
+const showBrands = inventory.map((television) => {
+  return television.brand;
+})
+console.log(showBrands);
+
+const element4 = document.getElementById("testInnerHTML");
+element4.innerHTML =
+    `<p> De volgende merken zijn er: ${showBrands}</p>`;
+
+// // opdracht3b
+
+function tvsInList () {
+  let listOfBrands = [];
+  for (let i = 0; i < inventory.length; i++) {
+    listOfBrands[i] = inventory[i].brand;
+  }
+  console.log(listOfBrands);
+  const element5 = document.getElementById("listOfTelevisionBrands");
+  element5.textContent = `De volgende merknamen zijn er: ${listOfBrands}`;
+}
+
+tvsInList();
+
+
+//OPDRACHT 4
+//Opdracht 4a
+  function generateString (tvObject) {
+    return tvObject.brand + " " + tvObject.type + " - " + tvObject.name;
```

Comment:
Ik vind het altijd makkelijker te lezen en te begrijpen als je het op zo'n manier doet
```suggestion
    return `${tvObject.brand} ${tvObject.type - ${tvObject.name}`;
```